packages/logging/src/log-escalator.test.ts

```diff
@@ -0,0 +1,310 @@
+import pino from 'pino'
+import { logEscalator } from './log-escalator'
+
+it('does not write logs until done is called', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'info' },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+
+    expect(logs).toEqual([])
+})
+
+it('wraps a logger', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":10,"msg":"Test","v":1}
+`,
+        `{"level":20,"msg":"Test","v":1}
+`,
+        `{"level":30,"msg":"Test","v":1}
+`,
+        `{"level":40,"msg":"Test","v":1}
+`,
+        `{"level":50,"msg":"Test","v":1}
+`,
+        `{"level":60,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('done clears buffer', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.done()
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":10,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('does not support creating a child logger', () => {
```

Comment:
is there no way we can accommodate child loggers? we do use them in a few places. Not a deal breaker however. Errors coming with detailed logs will make up for it provided they have a correlation ID.

Can this logger provide a correlation ID by default?

Reply:
The idea is that the log escalator would be used on a child logger. Usage is:

`logEscalator(log.child({ requestId: ....})`